src/cli/main/cli.py

```diff
@@ -286,16 +277,13 @@ def _nix_build(
             ]
         )
     return [
-        *_if(NIX_STABLE, f"{__NIX_STABLE__}/bin/nix-build"),
-        *_if(not NIX_STABLE, f"{__NIX_UNSTABLE__}/bin/nix"),
-        *_if(not NIX_STABLE, "--experimental-features", "flakes nix-command"),
-        *_if(not NIX_STABLE, "build"),
-        *_if(NIX_STABLE, "--argstr", "makesSrc", __MAKES_SRC__),
-        *_if(NIX_STABLE, "--argstr", "projectSrc", head),
+        *[f"{__NIX__}/bin/nix-build"],
+        *["--argstr", "makesSrc", __MAKES_SRC__],
+        *["--argstr", "projectSrc", head],
         *["--argstr", "attrPaths", attr_paths],
-        *_if(NIX_STABLE, "--attr", attr),
+        *["--attr", attr],
         *["--option", "cores", "0"],
-        *_if(not NIX_STABLE, "--impure"),
+        *["--impure"],
```

Comment:
I think we should look for a better approach here instead of building impure derivations